lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -4696,9 +4696,10 @@ def get_multi_check_constraints(
             # "CHECK (((a > 1) AND (a < 5))) NOT VALID"
             # "CHECK (some_boolean_function(a))"
             # "CHECK (((a\n < 1)\n OR\n (a\n >= 5))\n)"
+            # "CHECK (a NOT NULL) NO INHERIT"
 
             m = re.match(
-                r"^CHECK *\((.+)\)( NOT VALID)?$", src, flags=re.DOTALL
+                r"^CHECK *\((.+)\)( NOT VALID)?( NO INHERIT)?$", src, flags=re.DOTALL
```

Comment:
can a constraint be both not valid and no inherit? are they always in this order?

Reply:
From the documentation I understood a constraint could be specified as both `NOT VALID` and `NO INHERIT` and only in this order – however after testing I see these can be specified in either order.

I'll work on addressing this.

Reply:
it's only an issue if the db returns to you the definition in different order.
PG does not store the original sql, so what's returning is re-generated by its internal representation and does not depend on the input sql (namely spaces / cast / parenthesis are different from the input sql)

Reply:
ok, thanks for testing!

would be nice to have a test for this. It could create a table using textual sql using the two form and reflect it verifying that both are detected correctly

Reply:
I'd be happy to add more tests – can you point me to an example?